cfep-12.md

```diff
@@ -0,0 +1,48 @@
+
+<table>
+<tr><td> Title </td><td> Removing packages that violate the terms of the source package</td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Isuru Fernando &lt;isuruf@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Feb 26, 2020</td></tr>
+<tr><td> Updated </td><td> Feb 26, 2020</td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+Source packages that we use to create conda packages have terms and conditions that allow
+us to redistribute the package in binary form. This CFEP codifies the conduct when these
+violations are reported.
+
+## Implementation
+
+When a violation is brought to the attention of the Core team, the feedstock maintenance
+team should be made aware by the Core team in a github issue on the feedstock and notified using the
+github handles of the maintenance team members.
+
+A week would be given to the maintenance team to respond to the claim and
+another week would be given to address the issue. The feedstock maintenance team
+may ask to extend the deadlines.
+
+If the issue is not addressed after the specified time period, the package will be moved
+to the broken label.
```

Comment:
If I'm understanding correctly, then I don't think that moving the package to the broken label truly addresses the redistribution violation. We need to **remove** all binaries from the conda-forge channel until the redistribution violation is addressed. Otherwise we, conda-forge, are still redistributing the conda package that is in violation of the source packages terms and conditions. The "broken" label doesn't actually protect conda-forge from a redistribution violation.

If I'm understanding correctly, then 41 needs to be updated to say that we're going to _remove_ the package from conda-forge.

Reply:
Yep. broken label is still a channel and the package is still available. This is probably the only case were a removal is the right procedure.

Reply:
From a legal stand point I agree with @ericdill, deleting the package is the only way to stop re-distribution. 

From a user's stand point moving to `broken` is more attractive as it still allows access to the package when re-creating a historic environment.

Perhaps a compromise is to move the package to `broken` by default but remove the package if the upstream developers request this action.  My though is that many developers would be happy to have the license violation corrected even if packages exist which have a violation and are marked as such.

Do other packaging communities (Debian, CentOS, Arch) remove packages from their repositories when a license violation is found or do they release new builds to address the issue?

Reply:
Moving things "only to broken" could also get us into more trouble as we show that we know that we are aware of an issue but didn't take appropriate action.

Reply:
Both @jjhelmus and @xhochy points goes back to what I asked above. Shouldn't we take action if, and only if, we get a take down notice? We can ask NumFOCUS about it.

Reply:
I suggested we consult a lawyer on gitter and still think we should do that.

Reply:
I recall that the decision to wait on a take down notice was taken after we spoke with NumFOCUS people and Anaconda. We can still ask NumFOCUS about. I don't remember why we ended up not asking them to forward that to the lawyer.

PS: who is out primary NumFOCUS contact these days?

Reply:
Debian has some documentation on their removal process, https://www.debian.org/doc/manuals/developers-reference/pkgs.en.html#removing-pkgs.  

From my reading package are deleted from un-released distributions, ( `unstable`, `experimental` and `stable`) but not from released distributions (`jessie`, `stretch`, `buster`).

Not sure how or if this should translate to conda-forge which has little idea of a "released distribution".  Perhaps if a package is included in miniforge it should not be removed?

Reply:
@jjhelmus that does not say anything about removal due license violations. (Or I missed while reading it.)

If I recall correctly Debian has a very pick way of including packages and license issues are solved before they make into a distribution.